src/js/index.js

```diff
@@ -56,38 +59,90 @@ RequestUtil.json = function (options = {}) {
         </Provider>),
         domElement);
     } else {
-      setTimeout(function () {
-        let routes = RouterUtil.buildRoutes(appRoutes.getRoutes());
-        let router = Router.run(routes, function (Handler, state) {
-          Config.setOverrides(state.query);
-          ReactDOM.render(
-            (<Provider store={PluginSDK.Store}>
-              <Handler state={state} />
-            </Provider>),
-            domElement);
+      let loadedResources = {
+        mesos: false
+      };
+      let mesosEvents = [MESOS_SUMMARY_CHANGE, MESOS_SUMMARY_REQUEST_ERROR];
+
+      // Responsible for checking if application is ready to render
+      function delayOrLoadApplication() {
+        // Check if we've loaded all resources first
+        let resources = Object.keys(loadedResources);
+        let allLoaded = resources.reduce(function (memo, resource) {
+          if (loadedResources[resource] === false) {
+            memo = false;
+          }
+
+          return memo;
+        }, true);
+
+        if (allLoaded === false) {
+          return;
```

Comment:
Don't we want to make sure we call this function again to check we're all loaded?
```
setTimeout(delayOrLoadApplication, msLeftOfDelay);
return;
```